Python/ceval.c

```diff
@@ -1644,15 +1652,21 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, PyFrameObject *f, int throwflag)
     if (PyDTrace_FUNCTION_ENTRY_ENABLED())
         dtrace_function_entry(f);
 
+    if (!PyCodeObject_IsWarmedUp(co)) {
```

Comment:
Will we ever want to use type hints instead of warmup to set an initial specialization?

Reply:
Perhaps, but it would take quite a bit of effort to get it so that the type hints are even accessible. An idea I've seen mentioned in a talk or paper about Facebook's HHVM is to have opcodes that indicate type hints (though I have no idea exactly how this would be used). A problem I could imagine with using type hints is that people might write `Iterable[T]` for an argument but call it with `list[str]` -- the type hint is not directly usable in such cases. OTOH there are plenty of type hints that are concrete types, from `bool` to `List[int]`.